sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Percentile.scala

```diff
@@ -81,7 +96,11 @@ case class Percentile(
       case arrayData: ArrayData => arrayData.toDoubleArray().toSeq
   }
 
-  override def children: Seq[Expression] = child :: percentageExpression :: Nil
+  override def children: Seq[Expression] = if (withFrqExpr) {
```

Comment:
Let's make the param sequence to: `child, percentageExpression, frequencyExpression`, and give a default value to `frequencyExpression`.